repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,254 @@
+        console.print("Please specify metadata url")
+        return
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if not os.path.isfile(f"{metadata_dir}/root.json"):
+        console.print(
+            "Trusted local root not found. Using 'tofu' to "
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url} - "
+                "`tofu` was not successful"
+            )
+            return False
+
+    print(f"Using trusted root in {metadata_dir}")
+
+    if directory_prefix:
+        download_dir = directory_prefix
+    else:
+        download_dir = os.getcwd() + "/downloads"
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+        )
+        updater.refresh()
+
+        info = updater.get_targetinfo(artifact_name)
+
+        if info is None:
+            console.print(f"Artifact {artifact_name} not found")
+            return False
+
+        path = updater.find_cached_target(info)
+        if path:
+            console.print(f"Artifact is available in {path}")
+            return True
+
+        path = updater.download_target(info)
+        console.print(f"Artifact downloaded and available in {path}")
+
+    except (OSError, RepositoryError, DownloadError) as e:
+        console.print(f"Failed to download artifact {artifact_name}: {e}")
+        return False
+
+    return True
+
+
+@artifact.command()
+@click.option(
+    "-r",
+    "--root",
+    help="A metadata URL to the initial trusted root or a local file.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-m",
+    "--metadata-url",
+    help="TUF Metadata repository URL.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-a",
+    "--artifacts-url",
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A hash prefix.",
+    is_flag=True,
+    required=None,
+)
+@click.option(
+    "-P",
+    "--directory-prefix",
+    help="A prefix for the download dir.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.argument("artifact_name")
+@click.pass_context
+def download(
+    context: Context,
+    root: str,
+    metadata_url: str,
+    artifacts_url: str,
+    hash_prefix: bool,
+    directory_prefix: str,
+    artifact_name,
+) -> None:
+    """
+    Downloads artifacts to the TUF metadata.
+    """
+
+    rstuf_config = context.obj.get("settings")
+
+    if context.obj.get("config") and os.path.isfile(context.obj.get("config")):
```

Comment:
What happens if this initla `if` is not true?
I don't see any warnings about this and it will result with `AttributeError` on line 243.

In your place, I would rather do:
```suggestion
    if not context.obj.get("config") or not os.path.isfile(context.obj.get("config")):
        raise click.ClickException("Please specify current repository")
```
and I won't think about this case anymore and all `if` statements will shift to the left.

Reply:
It's not considered to fail or raise exceptions. What we do in this `if` (clarified in a comment as well) is to only overwrite the existing config with the corresponding flags when they are passed. What happens if it's False - the user is just not using a config file, so nothing to overwrite.

The reason there's no `ClickException` is that `CURRENT_REPOSITORY` is only a configuration requirement and designed for such use-cases. It's not supported by the prompt, as if you don't use a config file, you specify only a single repository details to refer to. With a config you can list as many repositories, as you wish, but you must point to the one you're using.

Reply:
I missed that you handle the required attributes in `download_artifact` and that's why I thought you should handle the case the config is missing as well.
I left some comments there.